lib/src/rules/avoid_web_libraries_in_flutter.dart

```diff
@@ -52,33 +40,82 @@ class AvoidWebLibrariesInFlutter extends LintRule implements NodeLintRule {
   @override
   void registerNodeProcessors(
       NodeLintRegistry registry, LinterContext context) {
-    final visitor = _Visitor(this);
+    final visitor = _Visitor(this, context);
     registry.addCompilationUnit(this, visitor);
     registry.addImportDirective(this, visitor);
   }
 }
 
+class Entry {
+  final File file;
+  YamlMap yaml;
+  int timestamp;
+  Entry(this.file) {
+    update();
+  }
+
+  void update() {
+    timestamp = file.modificationStamp;
+    final content = file.readAsStringSync();
+    yaml = _parseYaml(content);
+  }
+
+  static YamlMap _parseYaml(String content) {
+    try {
+      final doc = loadYamlNode(content);
+      if (doc is YamlMap) {
+        return doc;
+      }
+      // ignore: avoid_catches_without_on_clauses
+    } catch (_) {
+      // Fall-through.
+    }
+    return YamlMap();
+  }
+
+}
+
+class Pubspec {
+  static Map<String, Entry> cache = <String, Entry>{};
+
+  static YamlMap forUnit(CompilationUnit unit, LinterContext context) {
+    final filePath = path.join(context.package.root, 'pubspec.yaml');
+    final entry = cache[filePath];
+    if (entry != null) {
+      if (entry.file.modificationStamp != entry.timestamp) {
```

Comment:
Getting the modification stamp is not as inexpensive as you might hope, and this gets it twice: once here and once in `update`.